README.rst

```diff
@@ -258,46 +258,62 @@ For example:
 How To Upgrade Schema
 =====================
 
+We are using Alembic framework to facility schema migration. For more details,
+see https://alembic.sqlalchemy.org/en/latest/tutorial.html.
+
 Here's an example of a normal workflow for performing schema update.
 
-1. create a new changeset file in *pint_server/pint_db_migrate/versions/*. The
-   new changeset file must have the following format: *<d><d><d>_<string>.py*.
-   The first three digit of the filename is the version number, follow by an
-   underscore and a meaningful name of the changeset. The new changeset must
-   have the highest version number, which is usually a plus one increment of
-   the last version. For example, if *pint_server/pint_db_migrate/versions/*
-   currently contains a file *001_in_the_beginning.py*. The next changeset
-   should starts with *002*. Say if we want to add a new column *foo* to the
-   *amazonimages* table. We should create a new file named *002_add_foo_column_to_amazonimages.py* with the following content:
+1. create the Python 3.6 development virtual environment
 
    .. code-block::
 
-     from sqlalchemy import Table, MetaData, String, Column
+     ./bin/create_dev_venv.sh
 
-     def upgrade(migrate_engine):
-         meta = MetaData(bind=migrate_engine)
-         amazonimages = Table('amazonimages', meta, autoload=True)
-         foo = Column('foo', String(100))
-         foo.create(amazonimages)
+2. activate the development virtual environment
 
-     def downgrade(migrate_engine):
-         meta = MetaData(bind=migrate_engine)
-         amazonimages = Table('amazonimages', meta, autoload=True)
-         amazonimages.c.foo.drop()
+   .. code-block::
+
+     source dev_venv/bin/activate
 
-2. create the Python 3.6 development virtual environment
+3. update `pint_server/models.py` to reflect the latest changes
+
+4. if schema change is pretty vanilla and straight forward (i.e.
+   add a new table and add/remove non-indexing columns) you can just let
+   Alembic generate the change script for you. Otherwise, skip to the next
+   step. To do that, you first need to edit `pint_server/alembic.ini` file to
```

Comment:
See my comments below about instead providing a `pint_server/alembic.ini.example` that should be copied to `pint_server/alembic.ini`, which itself can be a .gitignore'd file, to aid in avoiding concerns about people committing credentials in PRs..

Also the `sqlalchemy.url` setting should have some pre-amble comments explaining the construction of the setting value.

Reply:
Good idea!

Reply:
Done